shared/triage_metadata.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2019 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+//go:generate mockgen -destination sharedtest/triage_metadata_mock.go -package sharedtest github.com/web-platform-tests/wpt.fyi/shared TriageMetadataInterface
+
+package shared
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/google/go-github/v28/github"
+	"gopkg.in/yaml.v2"
+)
+
+// TriageMetadataInterface encapsulates triage() method for testing purpose.
+type TriageMetadataInterface interface {
+	Triage(metadata MetadataResults) error
+}
+
+type triageMetadata struct {
+	ctx context.Context
+	metadataGithub
+	logger     Logger
+	httpClient *http.Client
+}
+
+type metadataGithub struct {
+	githubClient *github.Client
+	authorName   string
+	authorEmail  string
+	wptmetadataGitHubInfo
+}
+
+type wptmetadataGitHubInfo struct {
+	sourceOwner   string
+	sourceRepo    string
+	commitMessage string
+	commitBranch  string
+	baseBranch    string
+	prRepoOwner   string
+	prRepo        string
+	prBranch      string
+	prSubject     string
+	prDescription string
+}
+
+func getNewCommitBranchName(ctx context.Context, client *github.Client, sourceOwner string, sourceRepo string) string {
+	commitBranch := "auto-triage-branch" + generateRandomInt()
+	// If the commitBranch name already exist, genereate a new one. We consider that an error means the branch has not been found and needs to
```

Comment:
```suggestion
	// If the commitBranch name already exists, generate a new one. We consider an error to mean that the branch has not been found and needs to
```

Also nit; can you split this comment a little earlier? I know this is golang and so line length is... err... optional, but since we're already splitting for 'be created' here maybe we could split further back and get a more readable split. (E.g. split somewhere around 'branch')